rai/secure/ledger.cpp

```diff
@@ -217,7 +217,7 @@ void ledger_processor::state_block (rai::state_block const & block_a)
 void ledger_processor::state_block_impl (rai::state_block const & block_a)
 {
 	auto hash (block_a.hash ());
-	auto existing (ledger.store.block_exists (transaction, hash));
+	auto existing (ledger.store.block_exists (transaction, block_a.type (), hash));
```

Comment:
Wondering if we can directly put types here & for other similar ledger_processor items like rai::block_type::state

Reply:
Yeah, I couldn't decide on that :)

Reply:
I think it'll be okay -- we can always change it if it shows up in profiling